tests/rptest/services/redpanda.py

```diff
@@ -1767,12 +1767,79 @@ def all_up(self) -> bool:
 
     def metrics(
             self,
-            node,
+            pod_name: str = None,
             metrics_endpoint: MetricsEndpoint = MetricsEndpoint.PUBLIC_METRICS
     ):
-        text = self._cloud_cluster.get_public_metrics()
+        if metrics_endpoint == MetricsEndpoint.PUBLIC_METRICS:
+            text = self._cloud_cluster.get_public_metrics()
+        else:
+            text = self.__kubectl.exec(
+                'curl -f -s -S http://localhost:9644/metrics',
+                pod_name).decode()
         return text_string_to_metric_families(text)
 
+    def _extract_samples(self, metrics, sample_pattern: str,
+                         pod) -> list[MetricSamples]:
+        found_sample = None
+        sample_values = []
+
+        for family in metrics:
+            for sample in family.samples:
+                if sample_pattern not in sample.name:
+                    continue
+                if not found_sample:
+                    found_sample = (family.name, sample.name)
+                if found_sample != (family.name, sample.name):
+                    raise Exception(
+                        f"More than one metric matched '{sample_pattern}'. Found {found_sample} and {(family.name, sample.name)}"
+                    )
+                sample_values.append(
+                    MetricSample(family.name, sample.name, pod, sample.value,
+                                 sample.labels))
+
+        return sample_values
+
+    def metrics_sample(
```

Comment:
This is mostly just a copy/paste from RedpandaService, right? That's something we want to avoid: we should identify the little part which needs to be different between the two (the actual query part) and ensure we share the rest of the implementation. So for example all this could be hoisted into the ABC with the part that varies extracted out (`self.metrics()`?) and implemented by each service.

This means we need to provide a consistent interface too, e.g., both using "broker IDs" not "pod names" since the latter concept doesn't exist in vanilla installs.

Reply:
the code has been refactored to have abstract method `metrics_samples()` where the implementations use `_metrics_samples()` for the common work.